builder/src/lib.rs

```diff
@@ -160,10 +160,14 @@ pub fn elf2rom(elf_bytes: &[u8]) -> io::Result<Vec<u8>> {
         let Some(src_bytes) = elf_bytes.get(file_offset..file_offset + len) else {
             return Err(other_err(format!("segment at 0x{:x} out of file bounds", segment.p_offset)));
         };
-        let Some(dest_bytes) = result.get_mut(mem_offset..mem_offset + len) else {
-            continue;
-        };
-        dest_bytes.copy_from_slice(src_bytes);
+        if len > 0 {
```

Comment:
Optional: could do,

```
let len = segment.p_filesz as usize;
if len == 0 {
  continue;
}
let Some(src_bytes) = ...;
let Some(dest_bytes) = ...;
```
I assume the check against zero is needed because there are some out-of-bounds but zero-length segments that we don't want to trip over unnecessarily.